src/relax/op/tensor/manipulate.cc

```diff
@@ -421,5 +421,285 @@ TVM_REGISTER_OP("relax.flatten")
+  if (output_ndim == kUnknownNDim) {
+    return tensor_sinfo.size() == 1 ? tensor_sinfo[0] : TensorStructInfo(output_dtype, output_ndim);
+  }
+
+  int axis =
+      attrs->axis.defined() ? NormalizeAxis(call, ctx, output_ndim, attrs->axis.value()->value) : 0;
+  // If there is only one input tensor, no action is needed.
+  if (tensor_sinfo.size() == 1) {
+    return tensor_sinfo[0];
+  }
+  if (shape_values.empty()) {
+    return TensorStructInfo(output_dtype, output_ndim);
+  }
+
+  // As long as the there is known shape value, we will do the best effort check to ensure safety.
+  Optional<Array<PrimExpr>> output_shape = CheckComputeOutputShape(call, ctx, shape_values, axis);
+
+  if (shape_unknown || !output_shape.defined()) {
+    return TensorStructInfo(output_dtype, output_ndim);
+  } else {
+    return TensorStructInfo(ShapeExpr(output_shape.value()), output_dtype);
+  }
+}
+
+TVM_REGISTER_OP("relax.concat")
+    .set_attrs_type<ConcatAttrs>()
+    .set_num_inputs(1)
+    .add_argument("data", "Tuple of Tensors", "The input list of tensors.")
+    .set_attr<FInferStructInfo>("FInferStructInfo", InferStructInfoConcat);
+
+/* relax.split */
+TVM_REGISTER_NODE_TYPE(SplitAttrs);
+
+Expr MakeSplit(Expr data, ObjectRef indices_or_sections, int axis) {
+  ObjectPtr<SplitAttrs> attrs = make_object<SplitAttrs>();
+  attrs->indices_or_sections = indices_or_sections;
+  if (!indices_or_sections->IsInstance<ArrayNode>()) {
+    const auto* n_section = indices_or_sections.as<IntImmNode>();
+    CHECK(n_section != nullptr) << "Split op expects the input indices_or_sections to be either an "
+                                   "Array of PrimExpr or an integer. However, the given one is "
+                                << indices_or_sections->GetTypeKey();
+    CHECK_GT(n_section->value, 0) << "Split op expects the input number of sections to be a "
+                                     "positive integer. However, the given number of sections is "
+                                  << n_section->value;
+  }
+  attrs->axis = axis;
+
+  static const Op& op = Op::Get("relax.split");
+  return Call(op, {std::move(data)}, Attrs(attrs), {});
+}
+
+TVM_REGISTER_GLOBAL("relax.op.split").set_body_typed(MakeSplit);
+
+StructInfo InferStructInfoSplit(const Call& call, const BlockBuilder& ctx) {
+  TensorStructInfo data_sinfo = GetUnaryInputTensorStructInfo(call, ctx);
+  const auto* attrs = call->attrs.as<SplitAttrs>();
+  const auto* data_shape = data_sinfo->shape.as<ShapeExprNode>();
+  int axis = data_sinfo->IsUnknownNdim() ? attrs->axis
+                                         : NormalizeAxis(call, ctx, data_sinfo->ndim, attrs->axis);
+
+  if (const auto* p_indices = attrs->indices_or_sections.as<ArrayNode>()) {
+    if (p_indices->size() == 0) {
+      return TupleStructInfo({data_sinfo});
+    }
+    for (int i = 0; i < static_cast<int>(p_indices->size()); ++i) {
+      ObjectRef idx = p_indices->at(i);
+      const auto* p_idx = idx.as<PrimExprNode>();
```

Comment:
Given indices_or_sections is in Attrs, do not allow symbolic expression for now, only allow constant int

Reply:
Okay I think this is good for now, and we can revisit the Split API in the future.